app/src/main/java/com/arygm/quickfix/ui/authentication/RegisterScreen.kt

```diff
@@ -265,7 +266,9 @@ fun RegisterScreen(
                                       onValueChange = {
                                         email = it
                                         emailError = !isValidEmail(it)
-                                        profileViewModel.profileExists(email) { exists, profile ->
+                                        profileViewModel.profileExists(ProfileType.USER, email) {
+                                            exists,
+                                            profile ->
                                           emailError =
                                               if (exists && profile != null) {
                                                 true
```

Comment:
I believe this version is cleaner:

`emailError = exists && profile != null || !isValidEmail(it)`